R/features.R

```diff
@@ -414,6 +414,337 @@ setMethod("calculatePeakQualities", "features", function(obj, weights, flatnessF
+  
+  filePaths <- getMzMLOrMzXMLAnalysisPath(anaInfo$analysis, anaInfo$path, mustExist = TRUE)
+  
+  res <- lapply(filePaths, function(fpath)
+  {
+    hd <- getHeaders(fpath, retentionRange, msLevels)
+    data.table("ret" = hd$retentionTime, "msLevel" = hd$msLevel, "intensity" = hd$totIonCurrent)
+  })
+  
+  names(res) <- anaInfo$analysis
+  group <- anaInfo$group
+  names(group) <- anaInfo$analysis
+  res <- rbindlist(res, idcol = "analysis")
+  res$group <- group[res$analysis]
+  setcolorder(res, c("analysis", "group"))
+  res
+})
+
+
+#' @describeIn features Obtain the base peak chromatogram/s (BPCs) of the analyses.
+#' @aliases getBPCs
+#' @export
+setMethod("getBPCs", "features", function(obj, i = NULL, retentionRange = NULL, msLevels = 1)
+{
+  ac <- checkmate::makeAssertCollection()
+  
+  msLevels <- as.integer(msLevels)
+  checkmate::assert(checkmate::check_integer(msLevels, min.len = 1, lower = 1), add = ac)
+  
+  anaInfo <- analysisInfo(obj)
+  
+  if (!is.null(i))
+  {
+    if (is.character(i)) {
+      checkmate::assert(checkmate::check_choice(i, anaInfo$analysis), add = ac)
+      anaInfo <- anaInfo[anaInfo$analysis %in% i, ]
+    } else if (is.integer(i)) {
+      i <- as.integer(i)
+      checkmate::assert(checkmate::check_integer(i, min.len = 1, lower = 1, upper = nrow(anaInfo), max.len = nrow(anaInfo)), add = ac)
+      anaInfo <- anaInfo[i, ]
+    }
+  }
+  
+  checkmate::reportAssertions(ac)
+  
+  filePaths <- getMzMLOrMzXMLAnalysisPath(anaInfo$analysis, anaInfo$path, mustExist = TRUE)
+  
+  res <- lapply(filePaths, function(fpath)
+  {
+    hd <- getHeaders(fpath, retentionRange, msLevels)
+    data.table("ret" = hd$retentionTime, "msLevel" = hd$msLevel, "mz" = hd$basePeakMZ, "intensity" = hd$basePeakIntensity)
+  })
+  
+  names(res) <- anaInfo$analysis
+  group <- anaInfo$group
+  names(group) <- anaInfo$analysis
+  res <- rbindlist(res, idcol = "analysis")
+  res$group <- group[res$analysis]
+  setcolorder(res, c("analysis", "group"))
+  res
+})
+
+
+#' @describeIn features Plots the base peak chromatogram/s (BPCs) of the analyses.
+#' @param retMin Plot retention time in minutes (instead of seconds).
+#' @param title Character string used for title of the plot. If \code{NULL} a title will be automatically generated.
+#' @param colourBy Sets the automatic colour selection: "none" for a single 
+#' colour or "analyses"/"replicates" for a distinct colour per replicate/analysis.
+#' @param showLegend Plot a legend if TRUE.
+#' @template plot-lim
+#' @aliases plotTICs
+#' @export
+setMethod("plotTICs", "features", function(obj,
+                                           i = NULL,
+                                           retentionRange = NULL,
+                                           msLevels = 1,
+                                           retMin = FALSE,
+                                           title = NULL,
+                                           colourBy = c("none", "analyses", "replicates"),
```

Comment:
Can you change `"replicates"` to `"rGroups"`? Again, mainly for consistency with other code, even if `"replicates"` would've been nicer...

Reply:
Changed